run_querytns.py

```diff
@@ -0,0 +1,233 @@
+&redshift_max={}&hostname=&ext_catid=&ra_range_min=&ra_range_max=\
+&decl_range_min=&decl_range_max=&discovery_instrument[]=null\
+&classification_instrument[]=null&associated_groups[]=null\
+&at_rep_remarks=&class_rep_remarks=&frb_repeat=all&frb_repeater_of_objid=\
+&frb_measured_redshift=0&frb_dm_range_min=&frb_dm_range_max=&frb_rm_range_min=\
+&frb_rm_range_max=&frb_snr_range_min=&frb_snr_range_max=&frb_flux_range_min=\
+&frb_flux_range_max=&num_page=500&display[redshift]=1&display[hostname]=1\
+&display[host_redshift]=1&display[source_group_name]=1\
+&display[classifying_source_group_name]=1&display[discovering_instrument_name]=0\
+&display[classifing_instrument_name]=0&display[programs_name]=0\
+&display[internal_name]=1&display[isTNS_AT]=0&display[public]=1\
+&display[end_pop_period]=0&display[spectra_count]=1\
+&display[discoverymag]=1&display[discmagfilter]=1\
+&display[discoverydate]=1&display[discoverer]=1\
+&display[remarks]=0&display[sources]=0&display[bibcode]=0\
+&display[ext_catalogs]=0".format(months,objtype,date_begin,date_end,z_min,z_max)
+
+    logger.info('Date begin = {}, date_end = {}'.format(date_begin,date_end))
+    logger.debug(query)
+
+    # Query TNS
+    logger.info('querying TNS for all targets, this may take awhile')
+    con = requests.post(query)
+    con.raise_for_status()
+
+    logger.info('query successful with status code: {}'.format(con.status_code))
+    # Parse output using BS4, find html table
+    soup = BeautifulSoup(con.text, 'html.parser')
+    tab = soup.find("table",'results-table')
+    logger.info('HTML table found')
+
+    if tab is None:
+        logger.warn('No HTML table obtained from query!')
+        raise AttributeError('No HTML Table Found')
+
+    #Get SN names
+    nms = get_names(tab)
+
+    # Remove already existing names using flows api
+    included_names = []
+    flows_targets = api.get_targets()
+    for target in flows_targets:
+        included_names.append('SN'+target['target_name'])
+    nms = list(set(nms)-set(included_names))
+    logger.info('Target names obtained: {}'.format(nms))
+
+    # Query TNS for object info using API, then upload to FLOWS using API.
+    sleep_time = 0.25 # seconds to wait in between each query
+    for name in nms:
+        sn = name.replace('SN','')
+        logger.info('querying TNS for: {}'.format(sn))
+
+        # make GET request to TNS via API
+        get_obj = [("objname",sn), ("photometry","0"), ("spectra","0")]
+        response = get_obj_tns(url_tns_api,get_obj)
+
+        # Parse output
+        if None not in response:
+            response.raise_for_status()
+
+            parsed = response.json()
+            #parsed = json.loads(response.text,object_pairs_hook=OrderedDict)
+            logger.info('GET query successful')
+
+            # Extract object info
+            reply = parsed['data']['reply']
+            name = reply['objname']
+            coord = SkyCoord(ra=reply['radeg']*u.deg,dec=reply['decdeg']*u.deg)
```

Comment:
Still needs to specify which coordinate frame things are retuned in by TNS (FK5/ICRS?). This is important because the API will convert things to ICRS internally and therefore this could introduce coordinate shifts if not specified.